### ISSUE TITLE
2019Q4 - 4.1 Port Report Site Issue web extension to Android component

### ISSUE THREAD

**Issue by miketaylr** [mozilla/webcompat-team-okrs#110]
The current Fenix implementation just sends a URL to the form. Our Web Extension sends a screenshot, URL, console logs, and some preferences. 

The goal here is to port this to an Android component (similar to the work we did last quarter with the webcompat system addon) and ship it in Fenix, to replace the current hard-coded menu item.

**Comment by miketaylr** [comment]
Update: with Tom moving to GeckoView for the next 6 months, it's not super clear how we're going to pull this off this quarter.

**Comment by miketaylr** [comment]
Right now, we're blocked on missing functionality to make this a clean and easy port. 

We need the following APIs:

|  API  | Tracking Bug  |
|---|---|
|   [webextension experiments](webextensions-experiments.readthedocs.io)|  ?? |
|  [`tabs.create`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/create) |   https://github.com/mozilla-mobile/android-components/issues/4438 (done)|
|   [`browserAction.onclicked`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction/onClicked)|  https://bugzilla.mozilla.org/show_bug.cgi?id=1530402 & https://github.com/mozilla-mobile/android-components/issues/4469  | [`tabs.onActivated`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/onActivated)
|   [`tabs.query`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/query) | done |
| [`tabs.executeScript`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/executeScript) | ?? |
| [`tabs.captureVisibleTab`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/captureVisibleTab)| ?? |
| (probably?) [`menus.create`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/menus/create)|  ?? |

For easy bookkeeping, here's a link to where the last version of the Fennec Web Extension was removed: https://phabricator.services.mozilla.com/D51194#change-GtdqiEQWYBgp

@wisniewskit am I missing anything?

**Comment by wisniewskit** [comment]
I think that covers it. For experimental APIs, I believe the Fennec version was checking:
- whether parental controls are active
- whether the "view as desktop" mode was activated, to offer a notification (snackbar) to file an issue

In addition, the following info was fetched from appInfo/the chrome window for a given tab:
- the buildID, release channel, and whether webrender/graphics acceleration are on
- whether this is a private browsing tab or not
- the console log for that tab
- if there is any blocked mixed content or anything blocked by tracking protection (not sure if we need this anymore if we can parse it from the console log?)

**Comment by denschub** [comment]
I did some work that is building an empty webextension with a port into native code for some experimentations, but this is not yet further than the "oh yeah, it builds and doesn't crash on startup" phase yet.

**Comment by denschub** [comment]
There have been some chats and some experiments. I started working on the port itself, but didn't spent enough time yet to make a great update, because All Hands. :)

**Comment by denschub** [comment]
Worked around some UI issues last week, working around some POST issues this week. More WebExtension APIs work than I expected.

**Comment by denschub** [comment]
Got everything working in the reference browser with two exceptions:

* `about:config` prefs, as there are no prefs in Fenix
* the whole `tabExtras` APIs, which include console logging, mixed content, ... Currently working on maybe porting those as far as possible.

**Comment by miketaylr** [comment]
@denschub (for when you're back), are we ready to start landing this (and file bugs for future work)? wondering if we should try to capture some of the additional work in Q2 OKRs.

**Comment by denschub** [comment]
Got stuck on fixing a weird warning-spam in logcat and working through one last stupid issue with disabling the component for release users. Wanted to get that done last week, but got blocked by... brains and viruses. Will provide a meaningful update as soon as possible.
